### PR TITLE
feat(render): add wrapper component option

### PR DESCRIPTION
**Pull request by alexkrolick**
**What**:

Closes #296 - allows adding a wrapper component to `render` 

**Why**:

to make it easier to write custom render functions for common HOCs and data/context providers

**How**:

**Checklist**:

- [x] Documentation added to the [docs site](https://github.com/alexkrolick/testing-library-docs) https://github.com/alexkrolick/testing-library-docs/pull/39
- [x] Tests
- [x] Typescript definitions updated
- [x] Ready to be merged
      
- [x] Added myself to contributors table

### CODE REVIEW

File: typings/index.d.ts [hunk 1]
```diff
@@ -25,17 +25,16 @@ export type HookResult<TResult> = {
...
-export type HookOptions = {
-  wrapper: React.FunctionComponent
-}
-
 export interface RenderOptions<Q extends Queries = typeof queries> {
   container?: HTMLElement
   baseElement?: HTMLElement
   hydrate?: boolean
   queries?: Q
+  wrapper: React.ComponentType
```
Comment (gnapse) [inline typings/index.d.ts]: I think this should be `wrapper?: WrapperComponent`.

File: src/__tests__/render.js [hunk 2]
```diff
@@ -90,3 +90,24 @@ it('supports fragments', () => {
...
   expect(document.body.innerHTML).toBe('')
 })
+
+test('renders options.wrapper around node', () => {
+  const WrapperComponent = ({children}) => (
+    <div data-testid="wrapper">{children}</div>
+  )
+
+  const {container, getByTestId} = render(<div data-testid="inner" />, {
+    wrapper: WrapperComponent,
```
Comment (kentcdodds) [inline src/__tests__/render.js]: Should this be: `wrapper: ({children}) => <Wrapper>{children}</Wrapper>` so people have more flexibility?

Comment (alexkrolick) [reply src/__tests__/render.js]: You mean call it as a function instead of with createElement?

**Comment by kentcdodds** [comment]
Whoops, there's a merge conflict

**Comment by alexkrolick** [comment]
(Fixed the conflict)

**Comment by kentcdodds** [comment]
Thanks!

**Pull request merged**